### PR TITLE
Print pod events when pod waiting fails

### LINKED ISSUE #53: When installer times out waiting for a Pod, print the relevant events
E.g. when dockerhub is rate limiting us, carrier installer will simply timeout but there is not way for the user to tell because the relevant events are never printed out.

The user could go and check manually on what's wrong but it would be useful to see the relevant events immediately.

It's also useful for CI (when things timeout) so we don't need to find a way to ssh to the CI container and check.

## Possible fix:

We need to run the equivalent of this command:

```
kubectl get event --namespace carrier-registry --field-selector involvedObject.name=$(kubectl get pods -o=jsonpath='{.items[0].metadata.name}'  --selector=app.kubernetes.io/name=container-registry -n carrier-registry)
```

somewhere here: https://github.com/SUSE/carrier/blob/main/cli/kubernetes/cluster.go#L213

### PR DESCRIPTION
**Pull request by jimmykarily**
so the user has an idea of what went wrong (e.g. Dockerhub rate limiting)

You can test it by setting this to `1`: https://github.com/SUSE/carrier/blob/main/cli/cmd/install.go#L14

Fixes #53

### CODE REVIEW

File: cli/kubernetes/cluster.go [hunk 1]
```diff
@@ -210,12 +210,56 @@ func (c *Cluster) WaitForPodBySelectorRunning(namespace, selector string, timeou
...
+				return errors.Wrap(err, err2.Error())
+			} else {
+				return errors.New(fmt.Sprintf("Failed waiting for %s: %s\nPod Events: \n%s", pod.Name, err.Error(), events))
+			}
 		}
 	}
 	return nil
 }
 
+// GetPodEventsWithSelector tried to find a pod using the provided selector and
```
Comment (andreas-kupries) [inline cli/kubernetes/cluster.go:224]: `tried` -> `tries`

File: cli/kubernetes/cluster.go [hunk 2]
```diff
@@ -210,12 +210,56 @@ func (c *Cluster) WaitForPodBySelectorRunning(namespace, selector string, timeou
...
+// namespace. If found it returns the events on that Pod. If not found it returns
+// and error.
```
Comment (andreas-kupries) [inline cli/kubernetes/cluster.go:226]: `an error`

File: cli/kubernetes/cluster.go [hunk 3]
```diff
@@ -210,12 +210,56 @@ func (c *Cluster) WaitForPodBySelectorRunning(namespace, selector string, timeou
...
+// And equivalent kubectl command would look like this
```
Comment (andreas-kupries) [inline cli/kubernetes/cluster.go:227]: `An equivalent ...`

**Review by andreas-kupries — CHANGES_REQUESTED** [review]
Looks ok beyond the typo nits.

Question is, how can we test this, i.e. force a failure, to see that we get sensible information ?

File: cli/kubernetes/cluster.go (hunk 2 above)
Comment (jimmykarily) [reply cli/kubernetes/cluster.go:226]: fixed.

File: cli/kubernetes/cluster.go (hunk 3 above)
Comment (jimmykarily) [reply cli/kubernetes/cluster.go:227]: So many typos on one method description! :D Fixed.

**Comment by jimmykarily** [comment]
> Looks ok beyond the typo nits.
> 
> Question is, how can we test this, i.e. force a failure, to see that we get sensible information ?

Check the description of the PR. You can set the DefaultTimeout to 1 second which will make it fails almost certainly on some Pod. You should then be able to see the events (e.g. Downloading Image or something like that).

**Review by andreas-kupries — APPROVED** [review]

**Pull request merged**